File: 2238-a-number-after-a-double-reversal/solution.c
bool isSameAfterReversals(int num) {
    int rev = 0, rev2 = 0;
    int temp = num;
    while(temp != 0){
        rev = rev*10 + temp%10;
        temp/=10;
    }
    temp = rev;
    while(temp != 0){
        rev2 = rev2*10 + temp%10;
        temp/=10;
    }
    if(num == rev2) return true;
    else return false;
}

File: 2238-a-number-after-a-double-reversal/solution.java
class Solution {
    public int reverseNumber(int n){
        int rev = 0;
        while(n!=0){
            int rem = n %10;
            rev = rev * 10 + rem;
            n /= 10;
        }
        return rev;
    }
    public boolean isSameAfterReversals(int num) {
        int rev = reverseNumber(num);
        int again = reverseNumber(rev);
        if(again == num){
            return true;
        }
        return false;
    }
}
